### DISCUSSION TITLE
Systematic effort to make C wrappers?

### DISCUSSION THREAD

**Discussion by qpwo** [vlang/v#13589] [General]
How much effort has been dedicated, and how much effort would it take, to have proper V wrappers around all the major C libraries, so that a V user can pretend they're in V world for all major tasks?

I think this, together with [improving autofree](https://github.com/vlang/v/discussions/13588) would increase adoption dramatically.

**Comment by qpwo** [comment]
Consider efforts to rewrite the linux kernel in rust. It's too massive. So instead there are efforts to rewrite bits and pieces in rust, but integration between rust and C is honestly kind of poor. I think V would integrate much better. If modules of the linux kernel were wrapped in V then it would be much easier to replace  one module. But again, autofree would have to be near-perfect for that to be viable I think.

**Comment by JalonSolov** [comment]
One of the features of the `c2v` tool, once released, will be to output C wrappers instead of pure V translations.

It's not ready for release, yet, but it is being improved all the time.

As for "linux kernel"... take a look at the [vinix](https://github.com/vlang/vinix) project.

**Reply by qpwo** [reply]
Oh dope i didn't know something like the autowrappers was planned. That would basically instantly make V suitable for all would-be-C projects where a little bit of GC is acceptable.

I'm aware of vinix, very cool project, but smoothly integrating into the standard kernel would also be incredible

**Reply by ylluminate** [reply]
@qpwo you may be interested in this: https://github.com/spytheman/simple_kernel_module_in_v (shared: [Awesome V](https://github.com/vlang/awesome-v#operating-systems--os-development-examples))
